llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -68,6 +68,11 @@ static cl::opt<bool> EnableOrLikeSelectOpt("enable-aarch64-or-like-select",
 static cl::opt<bool> EnableLSRCostOpt("enable-aarch64-lsr-cost-opt",
                                       cl::init(true), cl::Hidden);
 
+static cl::opt<unsigned> SmallMultiExitLoopUF(
+    "small-multi-exit-loop-unroll-factor", cl::init(0), cl::Hidden,
```

Comment:
The name makes it sounds like a generally applicable flag, but it is something specific for the AArch64 cost model. I think ideally we would set things correctly per sub-target directly, which sounds like you are planning anyways, instead of adding this option first..